pkg/nodeagent/controller/operatingsystemconfig/containerd_config_test.go

```diff
@@ -0,0 +1,344 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package operatingsystemconfig_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/pelletier/go-toml"
+	"github.com/spf13/afero"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/utils/ptr"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/nodeagent/controller/operatingsystemconfig"
+	"github.com/gardener/gardener/pkg/utils/structuredmap"
+)
+
+var (
+	r   operatingsystemconfig.Reconciler
+	osc *extensionsv1alpha1.OperatingSystemConfig
+
+	ctx context.Context
+	log logr.Logger
+)
+
+const (
+	containerdConfigFilePath = "/etc/containerd/config.toml"
+
+	certdsDirValue    = "/etc/containerd/certs.d"
+	importsValue      = "/etc/containerd/conf.d/*.toml"
+	sandBoxImageValue = "foo:1.23"
+	cniBinDirValue    = "/opt/cni/bin"
+)
+
+func init() {
+	ctx = context.Background()
+	log = logr.Discard()
+
+	r = operatingsystemconfig.Reconciler{}
+
+	osc = &extensionsv1alpha1.OperatingSystemConfig{
+		Spec: extensionsv1alpha1.OperatingSystemConfigSpec{
+			CRIConfig: &extensionsv1alpha1.CRIConfig{
+				Name:         extensionsv1alpha1.CRINameContainerD,
+				CgroupDriver: ptr.To(extensionsv1alpha1.CgroupDriverSystemd),
+				Containerd: &extensionsv1alpha1.ContainerdConfig{
+					SandboxImage: sandBoxImageValue,
+				},
+			},
+			Units: []extensionsv1alpha1.Unit{},
+		},
+	}
+}
+
+func getMapEntry(m map[string]any, path structuredmap.Path) (any, error) {
+	if m == nil {
+		return nil, nil
+	}
+
+	key := path[0]
+
+	if len(path) == 1 {
+		return m[key], nil
+	}
+
+	entry, ok := m[key]
+	if !ok {
+		return nil, fmt.Errorf("unable to traverse into data structure because key %q does not exist", key)
+	}
+
+	childMap, ok := entry.(map[string]any)
+	if !ok {
+		return nil, fmt.Errorf("unable to traverse into data structure because value at %q is not a map", key)
+	}
+
+	r, err := getMapEntry(childMap, path[1:])
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
```

Comment:
I hope that the containerd configuration is not deeply structured so that this recursive approach causes problems, but then again this is just test code.

Reply:
This is pretty much directly taken from https://github.com/gardener/gardener/blob/e8880744dd5472fbb69686fd64a86ee47e59c3e8/pkg/utils/structuredmap/map.go#L30-L67. So far, the nesting in containerds config file has been rather shallow and this is used for tests only and not anywhere in production code (even though the `setMapEntry()` function which is used in production follows the same recursive approach). I do not see a huge problem with it in the test framework where we are providing the test data ourselves.